kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/DefaultKroxyliciousTester.java

```diff
@@ -147,14 +145,48 @@ public <U, V> Consumer<U, V> consumer(String virtualCluster, Serde<U> keySerde,
         return clients(virtualCluster).consumer(keySerde, valueSerde, additionalConfig);
     }
 
+    public void restartProxy() {
+        try {
+            proxy.close();
+            proxy = spawnProxy(kroxyliciousConfig);
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     @Override
     public void close() {
         try {
+            List<Exception> exceptions = new ArrayList<>();
+            for (KroxyliciousClients c : clients.values()) {
+                try {
+                    c.close();
+                }
+                catch (Exception e) {
+                    exceptions.add(e);
+                }
+            }
             proxy.close();
+            if (!exceptions.isEmpty()) {
+                // if we encountered any exceptions while closing, throw whichever one came first.
```

Comment:
I think we should log all of them though.

Reply:
That makes sense, have done that now